File: algorithm/24. Swap Nodes in Pairs/24.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        int first;  
        ListNode* phead = head;  
        while (head != nullptr && head->next != nullptr) {  
            first = head->val;  
            head->val = head->next->val;  
            head->next->val = first;  
            head = head->next->next;  
        }  
        return phead; 
    }
};

File: algorithm/24. Swap Nodes in Pairs/24.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        if( head == null)   return null;
        if(head.next == null)   return head;
        
        ListNode tem = head.next;
        head.next = swapPairs(tem.next);
        tem.next = head;
        return tem;
    }
}
